### ISSUE TITLE
"File does not end with a newline" false/erroneous diagnostic

### ISSUE THREAD

**Issue by slabstone** [nvuillam/npm-groovy-lint#440]
### Environment

Windows 10, Ubuntu 20.04 WSL and
```
$ nvm use --lts
Now using node v22.12.0 (npm v10.9.0)
$ npm exec -- npm-groovy-lint --version
GroovyLint: Successfully processed CodeNarc:
CodeNarc version 3.5.0
npm-groovy-lint version 15.0.2

Embeds:
CodeNarc version 3.5.0
- Groovy version 3.0.9 (superlite)
```

### Test file

```
echo "def s = 'hello'" > hello.groovy
```

`echo` adds newline but just it case let's check it:
```
$ xxd hello.groovy
00000000: 6465 6620 7320 3d20 2768 656c 6c6f 270a  def s = 'hello'.
```

newline character is present (`0a`)

### Run

```
npm exec -- npm-groovy-lint -c recommended-jenkinsfile -s hello.groovy
```

```
0
  0     info     File codeNarcTmpDir_0.7216443869861697.groovy does not end with a newline  FileEndsWithoutNewline


npm-groovy-lint results in 1 linted files:
┌─────────┬───────────┬─────────────┐
│ (index) │ Severity  │ Total found │
├─────────┼───────────┼─────────────┤
│ 0       │ 'Error'   │ 0           │
│ 1       │ 'Warning' │ 0           │
│ 2       │ 'Info'    │ 1           │
└─────────┴───────────┴─────────────┘
Failure: 0 error(s) have been found
 0 warning(s) have been found
 1 info(s) have been found
```

This message is unexpected
```
  0     info     File codeNarcTmpDir_0.7216443869861697.groovy does not end with a newline  FileEndsWithoutNewline
```
